contracts/token/ERC1155/base/ERC1155BaseInternal.sol

```diff
@@ -2,102 +2,31 @@
 
 pragma solidity ^0.8.0;
 
-import {IERC1155} from './IERC1155.sol';
-import {IERC1155Receiver} from './IERC1155Receiver.sol';
+import {AddressUtils} from '../../../utils/AddressUtils.sol';
+import {IERC1155Internal} from '../IERC1155Internal.sol';
+import {IERC1155Receiver} from '../IERC1155Receiver.sol';
 import {ERC1155BaseStorage} from './ERC1155BaseStorage.sol';
-import {AddressUtils} from '../../utils/AddressUtils.sol';
 
 /**
- * @title Base ERC1155 contract
+ * @title Base ERC1155 internal functions
  * @dev derived from https://github.com/OpenZeppelin/openzeppelin-contracts/ (MIT license)
  */
-abstract contract ERC1155Base is IERC1155 {
+abstract contract ERC1155BaseInternal is IERC1155Internal {
   using AddressUtils for address;
 
   /**
-   * @inheritdoc IERC1155
+   * @notice query the balance of given token held by given address
+   * @param account address to query
+   * @param id token to query
+   * @return token balance
    */
-  function balanceOf (
+  function _balanceOf (
     address account,
     uint id
-  ) virtual override public view returns (uint) {
-    require(account != address(0), 'ERC1155: balance query for the zero address');
```

Comment:
I probably would keep the require in the internal function, as querying the balance of the 0 address is probably a behavior that we would never want even in the internal function, and check is cheap anyway
(That particular error already helped us debugging the queue before)